evm/updateNonces.js

```diff
@@ -0,0 +1,74 @@
+'use strict';
+
+const chalk = require('chalk');
+const { Command, Option } = require('commander');
+const fs = require('fs');
+const { ethers } = require('hardhat');
+const { getDefaultProvider } = ethers;
+const readlineSync = require('readline-sync');
+
+const { printError, printInfo, printObj } = require('./utils');
+const { getNonceFromProvider, getAllSignersData } = require('./offline-sign-utils');
+
+function updateNonce(provider, addresses, filePath) {
+    try {
```

Comment:
`nonces.json` or whatever filePath provided can have a schema like
`{ env: { chain: { signer: nonce } } }`

signer addresses can be passed in as done below